src/lib/containers/MarkdownSynapse.js

```diff
@@ -233,13 +273,27 @@ class MarkdownSynapse extends React.Component {
         let count = 1
         let markup = this.createMarkup(this.state.text).__html.replace(/<span id="wikiReference.*?<span data-widgetparams.*?span>/g, 
             () => {
+                // replace all reference tags with id's that we can later target
                 let current = count++
                 return `<a href="" id="ref${current}">[${current}]</a>`
             }
         )
+
+        // for TOC
+        let tocHeadersRegex = /<h[1-6] toc="true">.*<\/h[1-6]>/gm
```

Comment:
constant (reused a couple of times).  if pattern needs to be changed in the future, there should be one place that the dev needs to change it.

Reply:
the two regexes are slightly different, the second is of the form:

```js
/<h([1-6]) id="(.*)" .*toc="true">(.*)<\/h[1-6]>/gm
```

I can still make them constants though